pkg/ecdsa/retry.go

```diff
@@ -0,0 +1,62 @@
+package ecdsa
+
+import (
+	"fmt"
+	"math/rand"
+	"sort"
+
+	"github.com/keep-network/keep-core/pkg/chain"
+)
+
+type byAddress []chain.Address
+
+func (ba byAddress) Len() int           { return len(ba) }
+func (ba byAddress) Swap(i, j int)      { ba[i], ba[j] = ba[j], ba[i] }
+func (ba byAddress) Less(i, j int) bool { return ba[i] < ba[j] }
+
+func EvaluateRetryParticipantsForSigning(
```

Comment:
Turns out that we are able to determine members who were inactive during the given attempt of DKG. Since inactivity will be much more common than actual malice, I'm wondering whether we can use the inactivity information during the retry algorithm somehow in order to increase the success rate. Any thoughts about that?

Reply:
just to be clear - this PR is about signing; the follow up will be about keygen!

If you have explicit inactivity information, here's what I would suggest (for both dkg and signing).

Step 1) exclude inactive participants and try to make it work. loop until the previous run failed but had full participation
Step 2) start using the retry algorithm from this PR on the participating operators. if operators start becoming inactive, go back to Step 1.

Reply:
Makes sense!